resources/user-agents/crawlers/search-engines/search-engines.json

```diff
@@ -474,13 +474,27 @@
           }
         },
         {
+          "match": "Mozilla/5.0 (compatible; Charlotte/*)",
+          "properties": {
+            "Browser": "Charlotte"
+          }
+        },
+        {
           "match": "Mozilla/5.0 (compatible; Charlotte/*; *)",
```

Comment:
I searched through our logs for useragents that match that first pattern, and found these:

>Mozilla/5.0 (compatible; Charlotte/1.1; http://www.searchme.com/support/)
Mozilla/5.0 (compatible; Charlotte/1.0t; http://www.searchme.com/support/)
Mozilla/5.0 (compatible; Charlotte/1.0b; charlotte@betaspider.com)
Mozilla/5.0 (compatible; Charlotte/0.9t; +http://www.searchme.com/support/)
Mozilla/5.0 (compatible; Charlotte/1.0b; http://www.searchme.com/support/)
Mozilla/5.0 (compatible; Charlotte/1.0b; http://www.betaspider.com/)

I'll add tests for these, but now I'm reconsidering adding a pattern for those Charlotte agents that were already in test case 900:

>Mozilla/5.0 (X11; U; Linux i686 (x86_64); en-US; rv:1.8.1.11) Gecko/20080109 (Charlotte/0.9t; http://www.searchme.com/support/)
Mozilla/5.0 (X11; U; Linux i686 (x86_64); en-US; rv:1.8.1.11) Gecko/20080109 (Charlotte/0.9t; http://www.searchme.com/support/) (Charlotte/0.9t; http://www.searchme.com/support/)

They are a different version, so perhaps the bot used a different UA format early in its life?

Thoughts?  Add to invalids test case, or put together a pattern that will catch these 2 as well as the others?